test/NuGet.Clients.Tests/NuGet.VisualStudio.Implementation.Test/Extensibility/VsPathContextProviderTests.cs

```diff
@@ -446,9 +450,8 @@ public async Task CreatePathContextAsync_WithUnrestoredPackageReference_Throws()
 
             var project = new TestPackageReferenceProject(projectUniqueName);
 
-            // Act
-            var exception = await Assert.ThrowsAsync<InvalidOperationException>(() => target.CreatePathContextAsync(project, CancellationToken.None));
-
+            // Assert
+            var exception = await Assert.ThrowsAsync<ProjectNotRestoredException>(() => target.CreatePathContextAsync(project, CancellationToken.None));
```

Comment:
Given the XMLDoc says that the exception is `InvalidOperationException`, I think that the tests should assert the contract, not the implementation, since tests should be validating the customer experience.

Unless the XMLDocs for the methods change, it's not a breaking change from a customer point of view, if a future refactor changes the exception to something other than `ProjectNotRestoredException`, as long as it's still extending `InvalidOperationException`.